internal/driver/driver.go

```diff
@@ -136,23 +165,56 @@ func (s *Driver) HandleWriteCommands(deviceName string, protocols map[string]mod
 		s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): deviceResourceName = %v", req.DeviceResourceName))
 		s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): protocols: %v, resource: %v, attribute: %v, parameters: %v", protocols, req.DeviceResourceName, req.Attributes, params))
 
-		key_type_value := fmt.Sprintf("%v", req.Attributes["type"])
+		// Get the value type from device profile
+		valueType := req.Type
+		s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): value type = %v", valueType))
 
+		var value interface{}
+		var err error
+
+		key_type_value := fmt.Sprintf("%v", req.Attributes["type"])
 		if key_type_value == "generic" {
-			if value, err := params[i].StringValue(); err == nil {
+			switch valueType {
+			case common.ValueTypeInt8:
+				value, err = params[i].Int8Value()
+				s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): Int8 = %v", value))
+			case common.ValueTypeInt16:
+				value, err = params[i].Int16Value()
+				s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): Int16 = %v", value))
+			case common.ValueTypeString:
+				value, err = params[i].StringValue()
+				s.lc.Debugf(fmt.Sprintf("Driver.HandleWriteCommands(): String = %v", value))
+			}
+
+			if err == nil {
 				key_timeout_value, _ := cast.ToIntE(req.Attributes["timeout"])
```

Comment:
Do not ignore error

Reply:
Should the timeout be a general custom coifing setting, rather that on every resource? 
Then  it is parsed on start-up and can avoid parsing it every time.

Reply:
Let's prefer it to be in device profile. It may change based on resource.